views/js/checkout-component-renderer.js

```diff
@@ -492,17 +497,49 @@ jQuery(document).ready(function() {
         }
 
         function handleOnClick(resolve, reject) {
-            const paymentMethodContainer = this.paymentForm.closest('.adyen-payment');
+            const paymentMethodContainer = getFormPaymentMethodContainer(this.paymentForm);
+            const paymentMethodType = paymentMethodContainer.data('local-payment-method');
             // Show message if button is disabled else if not in progress, hide and resolve
             if (prestaShopPlaceOrderButton.prop('disabled') && !isPlaceOrderInProgress()) {
                 showRequiredConditionsInfoMessage(paymentMethodContainer);
-                reject(new Error('Terms of service not agreed'));
+                if (paymentMethodType === 'paypal') {
+                    return false;
+                } else {
+                    reject(new Error('Terms of service not agreed'));
+                }
             } else if (!isPlaceOrderInProgress()) {
                 hideInfoMessage(paymentMethodContainer);
-                resolve();
+                if (paymentMethodType === 'paypal') {
+                    return true;
+                } else {
+                    resolve();
+                }
             }
         }
 
+        function handleOnCancel(state) {
```

Comment:
Doesn't the component exist in this callback here as a parameter?
I think you might not need to retrieve it with this `const component = getFormComponent(this.paymentForm);`

```suggestion
        function handleOnCancel(state, component) {
```